internal/pkg/table/adj.go

```diff
@@ -89,6 +89,43 @@ func (adj *AdjRib) Update(pathList []*Path) {
 	}
 }
 
+/* The provided pathList is expected to be the real candidate routes after policy evaluation.
+   For routes that are filtered by policy, there could be a mismatch between display
+   and actual rib sent to the peer (if softreset out was not run).
+   Only used to display adj-out because we do not maintain a separate adj-out table
+*/
+func (adj *AdjRib) UpdateAdjRibOut(pathList []*Path) {
```

Comment:
adj.Update() does the unnecessary work (handing withdraw, asloop check, etc) so you want to add this function that does only work necessary for adj-out? Or this fixes some bugs in Update() for  adj-out?

Reply:
The main difference is that adj.Update compares the paths based on PathIdentifier which is correct for adj-Rib-in whereas this function uses PathLocalIdentifier for matching existing paths which should be the behavior for adj-Rib-out.

Reply:
Ah, understood. Thanks!

Reply:
> PathLocalIdentifier for matching existing paths

This could actually happen?